protocols/Cargo.toml

```diff
@@ -2,7 +2,6 @@ name = "stratum_v2_protocols"
 version = "1.0.0"
 authors = ["The Stratum v2 Developers"]
 edition = "2021"
-rust-version = "1.75.0"
```

Comment:
I get the point of CI, but why is it desirable to remove it from the different `Cargo.toml` files?

Reply:
Yes, that has happened to me as well; my code editor continuously updates the `Cargo.lock` file because of this pinning. It would be nice to remove this pinning.

Reply:
the `README.md` already mention the MSRV, not sure about the hooks tbh as it seems its not widely used